docs/resources/rules/assertions.md

```diff
@@ -0,0 +1,471 @@
+- end with a _full stop_.
+
+In addition, the `Operation` summary property must:
+- be defined
+- be between 20 and 60 characters
+- not end with a _full stop_.
+
+The following example shows how to configure those assertions:
+
+```yaml
+lint:
+  rules:
+    assertions:
+      - subject: Tag
+        property: description
+        message: Tag description must be at least 30 characters and end with a full stop.
+        severity: error
+        defined: true
+        minLength: 30
+        pattern: /\.$/
+      - subject: Operation
+        property: description
+        message: Operation description must be at least 30 characters and end with a full stop.
+        severity: error
+        defined: true
+        minLength: 30
+        pattern: /\.$/
+      - subject: Info
+        property: description
+        message: Info description must be at least 30 characters and end with a full stop.
+        severity: error
+        defined: true
+        minLength: 30
+        pattern: /\.$/
+      - subject: Operation
+        property: summary
+        message: Operation summary must be between 20 and 60 characters and not end with a full stop.
+        severity: error
+        defined: true
+        minLength: 20
+        maxLength: 60
+        pattern: /[^\.]$/
+```
+
+### `property` example
+
+The following example asserts that every path item has a GET operation defined.
+
+```yaml
+lint:
+  rules:
+    assertions:
+      - subject: PathItem
+        property: get
+        message: Every path item must have a GET operation.
+        defined: true
+```
+
+A different way to declare the same assertion is to require that the `PathItem` has the `get` key.
+Notice we don't need to include `property` in this approach.
+
+```yaml
+lint:
+  rules:
+    assertions:
+      - subject: PathItem
+        message: Every path item must have a GET operation.
+        required: 
+          - get
+```
+
+The following example asserts that Tags have both name and description defined.
+
+```yaml
+lint:
+  rules:
+    assertions:
+      - subject: Tag
+        property:
+          - name
+          - description
+        message: Every tag must have a name and description.
+        defined: true
+```
+
+Another way to compose that rule is to require the subject keys:
+
+```yaml
+lint:
+  rules:
+    assertions:
+      - subject: Tag
+        message: Every tag must have a name and description.
+        required:
+          - name
+          - description
+```
+
+### `context` example
+
+The following example asserts that PUT responses with HTTP status 200 or 201 cannot return an `application/pdf`content type.
+Without the `context`, the assertion would evaluate every MediaTypeMap including:
+- Responses with all codes, including codes other than 200 or 201
+- Responses for all HTTP methods, including DELETE, GET, POST, and more.
+To restrict the evaluation, use the `context` feature to limit what will be evaluated.
+
+```yaml
+- context:
```

Comment:
issue: this example looks incomplete - it doesn't show where to place `context` in the `lint` section. It should be on the level of a single assertion, right?

Reply:
yes, it should be on the level of a single assertion, so I can add `assertions` at the top:
```suggestion
assertions:
  - context:
```

Reply:
already added assertions at the top